### PR TITLE
CI/CD integration with Azure integration tests

### PR DESCRIPTION
**Pull request by gonzalogarciajaubert**
## Description
* Enable "integration tests" for Azure into CI/CD

## Changes

* `.github/workflows/integration-tests.yml`
  * Add Azure Cli login and a new step to run Azure deploy/nuke tests with the expected env variables
* `.github/workflows/publish.yml`
  * Add Azure Cli login and a new step to run Azure deploy/nuke tests with the expected env variables
* `packages/framework-integration-tests/package.json`
  * Add Azure to the `integration` script. This will run the Azure integration tests on the publish GitHub action.
* packages/framework-provider-azure-infrastructure/src/infrastructure/helper/az-cli-helper.ts
  * Fix wrong log message

## Checks

- [x] Project Builds
- [x] Project passes tests and checks
- [x] Updated documentation accordingly (pending)

## Additional information

Azure integration tests took 39 minutes last time

### CODE REVIEW

File: .github/workflows/integration-tests.yml [hunk 1]
```diff
@@ -113,11 +116,65 @@ jobs:
...
+    steps:
+      - name: 'Az CLI login'
+        uses: azure/login@v1.4.0
+        with:
+          creds: '${{ secrets.AZURE_CREDENTIALS }}'
+
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v1
+        with:
+          node-version: 14.x
```
Comment (alvaroloes) [inline .github/workflows/integration-tests.yml:136]: I would use here 16 too, for consistency

File: .github/workflows/integration-tests.yml [hunk 2]
```diff
@@ -113,11 +116,65 @@ jobs:
...
```
Comment (alvaroloes) [inline .github/workflows/integration-tests.yml:165]: Same here, 16.x

**Review by alvaroloes — APPROVED** [review]
Merge :all-the-things:

File: .github/workflows/integration-tests.yml (hunk 1 above)
Comment (gonzalogarciajaubert) [reply .github/workflows/integration-tests.yml:136]: Good catch, I cherry-picked the changes and forgot to update this one. thanks

**Review by NickSeagull — APPROVED** [review]

**Review by javiertoledo — APPROVED** [review]

**Pull request merged**